### PR TITLE
[Translation] remove unused code

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 5.2
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Translation/Translator.php [hunk 1]
```diff
@@ -150,7 +150,7 @@ public function addResource(string $format, $resource, string $locale, string $d
     public function setLocale(string $locale)
     {
         $this->assertValidLocale($locale);
-        $this->locale = $locale ?? (class_exists(\Locale::class) ? \Locale::getDefault() : 'en');
+        $this->locale = $locale;
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Translation/Translator.php:153]: in 4.4, we deprecated calling setLocale with `null`

we could make the empty string fallback to `\Locale::getDefault()`, but there are tests that fail if we do.

File: src/Symfony/Component/Translation/Translator.php [hunk 2]
```diff
@@ -444,7 +444,7 @@ protected function computeFallbackLocales(string $locale)
      */
     protected function assertValidLocale(string $locale)
     {
-        if (null !== $locale && 1 !== preg_match('/^[a-z0-9@_\\.\\-]*$/i', $locale)) {
+        if (!preg_match('/^[a-z0-9@_\\.\\-]+$/i', $locale)) {
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Translation/Translator.php:447]: the empty string should not be a valid locale

**Review by derrabus — APPROVED** [review]

**Review by nicolas-grekas — COMMENTED** [review]
small mess here...

File: src/Symfony/Component/Translation/Translator.php (hunk 1 above)
Comment (ro0NL) [reply src/Symfony/Component/Translation/Translator.php:153]: ref #38230

we could still deprecate empty string on 4.4 isnt it? also the trait currenly stll allows `""` and `"0"`, perhaps we should consider a shared `LocaleAwareTrait`

Comment (nicolas-grekas) [reply src/Symfony/Component/Translation/Translator.php:153]: we can't deprecate anymore on 4.4
looks like we (I) missed why #38230 was proposed...

**Comment by nicolas-grekas** [comment]
Replaced by #42130

**Pull request closed without merging**